### PR TITLE
Add support for `idMatch`

### LINKED ISSUE #3: Add support for `idMatch`.
All calls to `Ember.deprecate` will include an `id` and `until` in Ember 2.0.0.  This addon can take advantage of that, to allow much better matching support (id's don't get dynamic interpolation).

### PR DESCRIPTION
**Pull request by gjastrab**
Can now match deprecations based on the `id` passed in the options
to `deprecate`.

Fixes #3.

### CODE REVIEW

File: tests/dummy/config/deprecation-workflow.js [hunk 1]
```diff
@@ -2,5 +2,6 @@ window.deprecationWorkflow = window.deprecationWorkflow || {};
 window.deprecationWorkflow.config = {"workflow":[
   { matchMessage: 'silence-me', handler: 'silence' },
   { matchMessage: 'log-me', handler: 'log' },
-  { matchMessage: 'throw-me', handler: 'throw' }
+  { matchMessage: 'throw-me', handler: 'throw' },
+  { idMatch: 'ember.workflow', handler: 'log' }
```
Comment (rwjblue) [inline tests/dummy/config/deprecation-workflow.js]: Let's tweak this to `matchId` (to keep with the same pattern as `matchMessage`).

**Comment by rwjblue** [comment]
Awesome, thanks for jumping in!  I'd like to see the option changed to `matchId` to better fit with `matchMessage`, but other than that this looks great!

**Comment by gjastrab** [comment]
Great - I almost did that but figured I'd follow as spelled in the issue first. I'll tweak that at lunch today and repush.

**Comment by gjastrab** [comment]
Amended and force pushed changing `idMatch -> matchId`

File: vendor/ember-cli-deprecation-workflow/main.js [hunk 2]
```diff
@@ -9,15 +9,18 @@
...
     for (i=0; i<config.workflow.length; i++) {
       workflow = config.workflow[i];
       matcher = workflow.matchMessage;
+      idMatcher = workflow.matchId;
 
       if (typeof matcher === 'string' && matcher === message) {
         return workflow;
       } else if (matcher instanceof RegExp && matcher.exec(message)) {
         return workflow;
+      } else if (typeof idMatcher === 'string' && options && idMatcher === options.id) {
```
Comment (rwjblue) [inline vendor/ember-cli-deprecation-workflow/main.js]: Lets make this the first check we do (I think `matchId` should "beat" `matchMessage`).

Comment (gjastrab) [inline vendor/ember-cli-deprecation-workflow/main.js]: Pushed - do you prefer PRs to be squashed down into single commits?

Comment (rwjblue) [inline vendor/ember-cli-deprecation-workflow/main.js]: Generally, yes I prefer them squashed but it isn't a big deal here...

**Comment by rwjblue** [comment]
Thank you!

**Pull request merged**